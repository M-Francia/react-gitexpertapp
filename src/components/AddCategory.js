import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => { 
    
    const [inputValue, setinputValue] = useState(''); //

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
    
        e.preventDefault(); //Para que no reviente

        if( inputValue.trim().length > 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setinputValue('');
        }

    }

    return(
        <form onSubmit={handleSubmit}>
        <h1>{inputValue}</h1>
        <p>{inputValue}</p>
        <input
            type="text"
            value={inputValue}
            onChange={ handleInputChange }
        />
        </form>
    )

 }

 AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}