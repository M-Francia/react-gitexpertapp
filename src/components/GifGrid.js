import React  from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifts } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


    const {data:images , loading} = useFetchGifs(category);

    //const [images, setImages] = useState([]);

    return (
    <>
        <h3>{category}</h3>
        
        <div className='card-grid'>
            {loading && <p>Loading...</p>}
                {
                    images.map( (img) =>{
                        return <GifGridItem
                            key={img.id} 
                            { ...img }
                        />
                    })
                }
        </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
