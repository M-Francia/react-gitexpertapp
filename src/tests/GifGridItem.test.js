import { shallow } from "enzyme"
import { GifGridItem } from "../components/GifGridItem"
import '@testing-library/jest-dom';


describe('GifGridItem', () => { 
    
    
    const title = 'titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should display de item', () => { 


        expect(wrapper).toMatchSnapshot();

     })


     test('should have <p></p> with the title', () => { 
        
        const  p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);

      })

      test('should have an image equal to url and alt from pros', () => { 
          
        const img = wrapper.find('img');

        
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);


       })

       test('should have animate_fadein', () => { 

        const div = wrapper.find('div');

        expect(div.prop('className')).toContain('animate__fadeIn');

        })

 })