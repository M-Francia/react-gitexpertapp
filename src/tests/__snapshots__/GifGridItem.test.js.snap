// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should display de item 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
